docs/guides/src/main/server/configuring-database.adoc

```diff
@@ -0,0 +1,92 @@
+
+== Configuring a database
+
+For each supported database, the server provides some opinionated defaults to make it simpler to configure a database. The defaults allow
+you to easily configure a database by just providing some key settings like the database host and credentials.
+
+To configure a database, run the `build` command to build a server image using the `db` option to configure the server for specific database:
+
+<@kc.build parameters="--db postgres --db-url-host mypostgres --db-username myuser --db-password change_me"/>
+
+In addition to the `db` configuration option, the `db-url-host`, `db-username`, and the `db-password`, are the bare minimum settings that you
+need to provide in order to successfully connect to a database.
+
+[NOTE]
+By default, the default schema is set to `keycloak`. You can change that by using the `db-schema` configuration option.
+
+== Configuring database credentials
+
+You should be able to set database credentials using the `db-username` and `db-password` configuration options.
+
+While tempting to set database credentials in plain text, you should avoid doing that whenever running in production in order to avoid unauthorized
+access to your database.
+
+The server allows you to securely store database credentials using different methods:
+
+* Using placeholders
+* Using the `build` command when creating a server image
+* Using Hashicorp Vault
+
+Some methods are more suitable when deploying to a specific environment. For instance, using placeholders and environment variables is suitable when running on Kubernetes.
+Each method has its pros and cons, and you should consider them carefully when picking one.
+
+=== Using placeholders to reference credentials
+
+Credentials can be resolved from environment variables when starting the server instead of having them explicitly set in
+a configuration file (e.g.: conf/keycloak.properties) or when passing arguments via CLI.
+
+<@kc.start parameters=r"--db-username ${MY_DB_USER} --db-password ${MY_DB_PASSWORD}"/>
+
+In this case, when the server is starting both ${r"MY_DB_USER"} and ${r"MY_DB_PASSWORD"} are going to be resolved as environment variables.
+
+=== Persisting credentials into the server image
+
+When running the `build` command, any configuration option you set is persisted into the server image. By having the database
+credentials into the server image, you no longer need to set them when starting the server.
+
+When using this method, you should make sure that:
+
+* The file permissions for the server distribution are properly set so that only specific users/groups are allowed to read and write to files
+* If using a container registry, make sure the repository is protected
+* The node where the server is running is properly secured from unathorized access
+
+=== Using a Vault
+
+You should be able to resolve the database credentials from Hashicorp Vault. For that, look at the guide https://link_to_guide[Configuring a Vault].
```

Comment:
Needs updating as we don't have Hashicorp initially. The vault guide also doesn't cover DB atm?